src/modules/kvs/kvs.c

```diff
@@ -729,8 +729,9 @@ static void get_request_cb (flux_t *h, flux_msg_handler_t *w,
          * Otherwise, use the current root.
          */
         if (root_dirent) {
-            if (j_dirent_validate (root_dirent) < 0
-                || !(root_ref = json_object_get (root_dirent, "DIRREF"))) {
+            if (treeobj_validate (root_dirent) < 0
+                || treeobj_is_dirref (root_dirent) < 0
```

Comment:
This was fixed in a prior commit, wondering why you didn't see it?

https://github.com/flux-framework/flux-core/pull/1154/commits/175ba7ad8751c0585cc7bb4bf88d28c5488ed214